src/tests/roc_audio/test_depacketizer.cpp

```diff
@@ -300,74 +301,136 @@ TEST(depacketizer, overlapping_packets) {
     expect_output(dp, SamplesPerPacket / 2, 0.33f);
 }
 
-TEST(depacketizer, frame_flags_dropped_packets_end) {
+TEST(depacketizer, frame_flags_packet_drops) {
     packet::ConcurrentQueue queue(0, false);
     Depacketizer dp(queue, pcm_decoder, ChMask, false);
 
-    const packet::timestamp_t ts1 = SamplesPerPacket * 3;
-    const packet::timestamp_t ts2 = SamplesPerPacket * 2;
-    const packet::timestamp_t ts3 = SamplesPerPacket * 1;
+    packet::PacketPtr packets[] = {
+        new_packet(SamplesPerPacket * 3, 0.11f),
+        new_packet(SamplesPerPacket * 1, 0.11f),
+        new_packet(SamplesPerPacket * 2, 0.11f),
+        new_packet(SamplesPerPacket * 6, 0.11f),
+        new_packet(SamplesPerPacket * 2, 0.11f),
+        new_packet(SamplesPerPacket * 3, 0.11f),
+    };
 
-    queue.write(new_packet(ts1, 0.11f));
-    queue.write(new_packet(ts2, 0.22f));
-    queue.write(new_packet(ts3, 0.33f));
+    unsigned frame_flags[] = {
+        Frame::FlagFull | !(Frame::FlagPacketDrops),
+        Frame::FlagPacketDrops | Frame::FlagEmpty,
+        Frame::FlagPacketDrops | Frame::FlagEmpty,
+        Frame::FlagFull | !(Frame::FlagPacketDrops),
+        Frame::FlagPacketDrops | Frame::FlagEmpty,
+        Frame::FlagPacketDrops | Frame::FlagEmpty,
+    };
 
-    Frame frame = new_frame(SamplesPerPacket * 3);
-    dp.read(frame);
+    CHECK(ROC_ARRAY_SIZE(packets) == ROC_ARRAY_SIZE(frame_flags));
+
+    for (size_t n = 0; n < ROC_ARRAY_SIZE(packets); n++) {
+        queue.write(packets[n]);
+    }
+
+    for (size_t n = 0; n < ROC_ARRAY_SIZE(packets); n++) {
+        Frame frame = new_frame(SamplesPerPacket);
+        dp.read(frame);
 
-    CHECK(frame.flags() & Frame::FlagSkip);
+        CHECK(frame.flags() & frame_flags[n]);
```

Comment:
Why do we need `&` here? Why not just listing the exact expected flags in `frame_flags` and then checking them here?

```
CHECK(frame.flags() == frame_flags[n]);
```

Reply:
I've fixed it.